newIDE/app/src/UI/TagChips.js

```diff
@@ -16,25 +17,76 @@ const styles = {
   },
 };
 
+const getChipColor = (tag: string) => {
+  return randomColor({
+    seed: tag,
+    luminosity: 'light',
+  });
+};
+
 type Props = {|
   tags: Tags,
   onChange?: Tags => void,
+  onRemove?: string => void,
 |};
 
-export default ({ tags, onChange }: Props) => {
+export default ({ tags, onChange, onRemove }: Props) => {
   if (!tags.length) return null;
 
+  const [focusedTag, setFocusedTag] = React.useState<?string>(null);
+  const [removedTagIndex, setRemovedTagIndex] = React.useState<number | null>(
+    null
+  );
+
+  const tagRefs = [];
+  React.useEffect(
+    () => {
+      if (removedTagIndex !== null) {
+        const tagToFocus = tagRefs[Math.min(removedTagIndex, tags.length - 1)];
+        tagToFocus.current && tagToFocus.current.focus();
+        setRemovedTagIndex(null);
+      }
+    },
+    [tags, removedTagIndex, tagRefs]
+  );
+
+  const getChipStyle = React.useCallback(
+    (tag: string) => ({
+      ...styles.chip,
+      backgroundColor:
+        !focusedTag || focusedTag !== tag ? undefined : getChipColor(tag),
```

Comment:
nit: you can compute `!focusedTag || focusedTag !== tag` only once before returning the object

Reply:
True, will change it